src/NJsonSchema/Generation/JsonSchemaGeneratorSettings.cs

```diff
@@ -80,6 +80,9 @@ public JsonSchemaGeneratorSettings()
         /// <summary>Gets or sets a value indicating whether to generate a description with number to enum name mappings (for integer enums only, default: false).</summary>
         public bool GenerateEnumMappingDescription { get; set; }
 
+        /// <summary>Will set `additionalProperties` on all added <see cref="JsonSchema">schema definitions and references</see>(default: false).</summary>
+        public bool AllowAdditionalProperties { get; set;}
```

Comment:
I'd call this AlwaysAllowAdditionalProperties because in the case of a dictionary (see generator) it is always true and cannot be switched off...

Reply:
Maybe also add a remark that this will break DTO code generation badly (I think)

Reply:
> 
> 
> You know that this will break client/DTO code generation completely? Because this is the only way to differentiate between dictionary and object

i have run all the tests and non failed. would you be able to provide test?

Reply:
i have tried to make the property nullable, so only of it was defined (set explicitly) than it was used, but that broke some tests (I guess tests which do settings reflection-serialization). 
so nullable could make clear and that option(true or false) is decided by other means until set explicitly. 

I will rename to `AlwaysAllowAdditionalProperties`, will add comment about inability to differentiate generation of C# code from that (not sure - may be will try to write tests for that). 

May be I will think if option at all needed for my case. I.e. prevent this kind of forward compatibility or adding new properties.

Reply:
If it's called AlwaysAllowAdditionalProperties then it should not be a nullable bool but just a bool.

It's just important to know that when this is enabled then code generation won't work as expected (might make sense if only json schema is required). Maybe it even works with this enabled - i hope we only generate a dictionary when no properties are defined... we should check that and maybe implement it...

Of course the tests do not fail because they test with AlwaysAllowAdditionalProperties = false => this is not a breaking change, it could just yield strange behavior when enabled.